### ISSUE TITLE
String Marshalling Issue

### ISSUE THREAD

**Issue by ds5678** [Speykious/NativeFileDialogs.NET#2]
On my machine (Windows 10, x64) the native strings seemed to be in Unicode instead of UTF8.

**Comment by Speykious** [comment]
Which example can illustrate the problem? Do you have screenshots?

**Comment by ds5678** [comment]
I was testing with the return strings from the save dialog examples, but it seems to be affecting all strings. Each ascii character was separated by a null terminator.

This caused `new string(pointer)` to always be of length 1.

**Comment by Speykious** [comment]
I'm on Linux but I'll investigate

**Comment by WiggleWizard** [comment]
Same issue; out path is always length of 1. Work around below:

```C#
public static unsafe bool NFDPickFolder(out string path, string defaultPath)
{
    const int buffSize = 1024; // Because of unicode, we can only capture `buffSize` / 2
    
    // Call into NFD to fetch the selected folder
    sbyte* outPathPtr;
    NfdresultT result = nfd.NFD_PickFolderN(&outPathPtr, defaultPath);
    
    // Copy the contents of the path from NFD into a buffer C# can understand (this is a Unicode buffer, so we copy 2
    // bytes at a time and stopping when the first byte of the next segment is zero)
    byte[] buffer = new byte[buffSize];
    int count = 0;
    for (int i = 0; i < buffSize - 1; i += 2)
    {
        if (*(outPathPtr + i) == 0)
        {
            count = i;
            break;
        }
        
        buffer[i + 0] = (byte)*(outPathPtr + i + 0);
        buffer[i + 1] = (byte)*(outPathPtr + i + 1);
    }
    
    // Unicode decode
    path = Encoding.Unicode.GetString(buffer, 0, count);
    
    return result == NfdresultT.NFD_OKAY;
}
```

Edit 1: needed to stop `GetString` from reading the entire buffer.
Edit 2: added comments and fixed a few things.

**Comment by Speykious** [comment]
Fixed in #3
